### PR TITLE
handwired 2x4 macropad

### PR DESCRIPTION
**Pull request by kubatyszko**
2x4 macropad, based on 2x5 

## Description

New keyboard - 2x4 macropad, using teensy

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/2x4keypad/2x4keypad.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include "2x4keypad.h"
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/2x4keypad.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/2x4keypad/2x4keypad.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/2x4keypad.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/2x4keypad/config.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/2x4keypad/config.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/config.h:37]:
```suggestion
```

File: keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c [hunk 5]
```diff
@@ -0,0 +1,164 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/2x4keypad/rules.mk [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/2x4keypad/2x4keypad.c (hunk 1 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/2x4keypad.c]: done

File: keyboards/handwired/2x4keypad/config.h (hunk 3 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/config.h:17]: done

File: keyboards/handwired/2x4keypad/config.h (hunk 4 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/config.h]: done

File: keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c (hunk 5 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c:17]: done

File: keyboards/handwired/2x4keypad/rules.mk (hunk 6 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/rules.mk]: done

File: keyboards/handwired/2x4keypad/2x4keypad.h (hunk 2 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/2x4keypad.h:17]: done

File: keyboards/handwired/2x4keypad/readme.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+active (no lit LEDs means the default layer).
+
+Keyboard Maintainer: [Kuba Tyszko](https://github.com/kubatyszko)  
+
+Hardware:
+  * 3d-Printed enclosure, STL here: https://www.thingiverse.com/thing:4671269
+  * Handwired
+  * Uses a Teensy
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/handwired/2x4keypad/readme.md:20]: Could you include instructions on how to jump into the bootloader?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/2x4keypad/readme.md (hunk 7 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/readme.md:20]: done.

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
+*/
+
+
```
Comment (fauxpark) [inline keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c:151]:
```suggestion
```

File: keyboards/handwired/2x4keypad/rules.mk [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
+BOOTMAGIC_ENABLE = full	# Virtual DIP switch configuration
+COMMAND_ENABLE = no     # Commands for debug and configuration
+CONSOLE_ENABLE= no	# Console for debug
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+MOUSEKEY_ENABLE = yes	# Mouse keys
+NKRO_ENABLE = yes	# USB Nkey Rollover -
+
+RGBLIGHT_ENABLE = no
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/handwired/2x4keypad/rules.mk:16]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/2x4keypad/rules.mk (hunk 9 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/rules.mk:16]: done

File: keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c (hunk 8 above)
Comment (kubatyszko) [reply keyboards/handwired/2x4keypad/keymaps/default_tapdance/keymap.c]: done

File: keyboards/handwired/2x4keypad/config.h [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+/* Tap dancing params */
+#define TAPPING_TERM 250
+
+/* key combination for command */
+/* DISABLED
+#define IS_COMMAND() ( \
+   get_mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+*/
+
```
Comment (fauxpark) [inline keyboards/handwired/2x4keypad/config.h:58]:
```suggestion
```

File: keyboards/handwired/2x4keypad/readme.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+    make handwired/2x4keyapd:default
```
Comment (fauxpark) [inline keyboards/handwired/2x4keypad/readme.md:22]:
```suggestion
    make handwired/2x4keypad:default
```

File: keyboards/handwired/2x4keypad/rules.mk [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+# Build Options
```
Comment (fauxpark) [inline keyboards/handwired/2x4keypad/rules.mk:6]:
```suggestion

# Build Options
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**